tuf/api/metadata.py

```diff
@@ -741,16 +741,24 @@ def to_dict(self) -> Dict[str, Any]:
 
     # Update key for a role.
     def add_key(self, role: str, key: Key) -> None:
-        """Adds new signing key for delegated role 'role'."""
+        """Adds new signing key for delegated role 'role'.
+
+        Raises:
+            ValueError: If 'role' doesn't exist.
+        """
+        if role not in self.roles:
+            raise ValueError(f"Role {role} doesn't exist")
         self.roles[role].keyids.add(key.keyid)
         self.keys[key.keyid] = key
 
     def remove_key(self, role: str, keyid: str) -> None:
         """Removes key from 'role' and updates the key store.
 
         Raises:
-            KeyError: If 'role' does not include the key
+            ValueError: If 'role' does not include the key.
         """
+        if keyid not in self.roles[role].keyids:
+            raise ValueError(f"Key with id {keyid} is not used by {role}")
```

Comment:
If you change the API, please make it match the new one you added for Targets. nonexisting role still raises a keyerror here and ValueError in Targets

Reply:
Yes, you are correct.
I didn't see that case before.